src/sentry/tasks/embeddings_grouping/utils.py

```diff
@@ -172,7 +172,7 @@ def get_current_batch_groups_from_postgres(
         group_id_filter, project.id, batch_size
     )
 
-    total_groups_to_backfill_length = len(groups_to_backfill_batch_raw)
+    total_groups_to_backfill_length = groups_to_backfill_batch_raw.count()
     batch_end_group_id = (
         groups_to_backfill_batch_raw[total_groups_to_backfill_length - 1][0]
         if total_groups_to_backfill_length
```

Comment:
You're now making two queries here - one to get the count, and one to get the end group id for the batch, in addition to fetching the batch on 183.

Is there any reason we can't just perform this logic after 183?

Any reason we can't move this logic after we filter to `groups_to_backfill_batch`?